plugins/base/src/main/kotlin/transformers/documentables/DefaultDocumentableMerger.kt

```diff
@@ -49,7 +51,7 @@ fun Function.mergeWith(other: Function): Function = Function(
     name,
     returnType,
     isConstructor,
-    if (receiver != null && other.receiver != null) receiver.mergeWith(other.receiver) else null,
+    (receiver to other.receiver).pullOutNull()?.let { (f, s) -> f.mergeWith(s) },
```

Comment:
Unnecessary overcomplication. You can simply go
`other.receiver?.let { receiver?.mergeWith(it) }`